frontend/src/views/CoursesList.js

```diff
@@ -34,15 +35,19 @@ class CoursesList extends Component {
       count: 0,
       pages: 0,
       currentPage: 1,
+      searchQuery: '',
+      ordering: 'display_name',
     };
 
     this.getCourses = this.getCourses.bind(this);
     this.setPerPage = this.setPerPage.bind(this);
+    this.setSearchQuery = this.setSearchQuery.bind(this);
+    this.setOrdering = this.setOrdering.bind(this);
   }
 
   getCourses(page = 1) {
     const offset = (page-1) * this.state.perPage;
-    const requestUrl = apiConfig.coursesGeneral + '?limit=' + this.state.perPage + '&offset=' + offset;
+    const requestUrl = apiConfig.coursesGeneral + '?search=' + this.state.searchQuery + '&ordering=' + this.state.ordering + '&limit=' + this.state.perPage + '&offset=' + offset;
```

Comment:
```suggestion
    const requestUrl = apiConfig.coursesGeneral + '?search=' + encodeURIComponent(this.state.searchQuery) + '&ordering=' + this.state.ordering + '&limit=' + this.state.perPage + '&offset=' + offset;
```

Reply:
Thanks for the suggestion, @OmarIthawi !